container/test-impl-base/src/main/java/org/jboss/arquillian/container/test/impl/client/deployment/AutomaticDeploymentScenarioGenerator.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2010 Red Hat Inc. and/or its affiliates and other contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.arquillian.container.test.impl.client.deployment;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Logger;
+import org.jboss.arquillian.container.spi.client.deployment.DeploymentDescription;
+import org.jboss.arquillian.container.spi.client.deployment.DeploymentScenario;
+import org.jboss.arquillian.container.spi.client.deployment.TargetDescription;
+import org.jboss.arquillian.container.spi.client.deployment.Validate;
+import org.jboss.arquillian.container.spi.client.protocol.ProtocolDescription;
+import org.jboss.arquillian.container.test.api.BeforeDeployment;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.DeploymentContent;
+import org.jboss.arquillian.container.test.api.ShouldThrowException;
+import org.jboss.arquillian.container.test.spi.client.deployment.AutomaticDeployment;
+import org.jboss.arquillian.container.test.spi.client.deployment.DeploymentScenarioGenerator;
+import org.jboss.arquillian.container.test.spi.util.ServiceLoader;
+import org.jboss.arquillian.test.spi.TestClass;
+import org.jboss.shrinkwrap.api.Archive;
+
+/**
+ * {@link DeploymentScenarioGenerator} that builds a {@link DeploymentScenario} based on
+ * the {@link AutomaticDeployment} registered services.
+ *
+ * @version $Revision: $
+ */
+public class AutomaticDeploymentScenarioGenerator extends AbstractDeploymentScenarioGenerator implements DeploymentScenarioGenerator {
+
+    AutomaticDeploymentLocator automaticDeploymentLocator;
+
+    public AutomaticDeploymentScenarioGenerator() {
+        automaticDeploymentLocator = new AutomaticDeploymentLocator() {
+            public ServiceLoader<AutomaticDeployment> find() {
```

Comment:
Why are we doing it like that? Why not inline it to `ServiceLoader.load(AutomaticDeployment.class)` in line [`63` ](https://github.com/arquillian/arquillian-core/pull/154/files#diff-c90c3d83ecfb089e296c0e97e93cff00R63)?

Reply:
Because it helps for unit testing. If not it means I need to create a  new Dummy service which affects all tests